python/python/lancedb/rerankers/linear_combination.py

```diff
@@ -95,43 +95,24 @@ def merge_results(
                     pa.array([nan] * len(vector_results), type=pa.float32()),
                 )
             return results
+        results = dict()
+        for vector_result in vector_results.to_pylist():
+            row_id = vector_result["_rowid"]
+            if row_id not in results:
+                results[row_id] = dict()
+            results[vector_result["_rowid"]]["_distance"] = vector_result["_distance"]
+        for fts_result in fts_results.to_pylist():
+            row_id = fts_result["_rowid"]
+            if row_id not in results:
+                results[row_id] = dict()
+            results[fts_result["_rowid"]]["_score"] = fts_result["_score"]
```

Comment:
I'd recommend [defaultdict](https://docs.python.org/3/library/collections.html#collections.defaultdict) here:
```suggestion
        results = defaultdict(dict)
        for vector_result in vector_results.to_pylist():
            row_id = vector_result["_rowid"]
            results[row_id]["_distance"] = vector_result["_distance"]
        for fts_result in fts_results.to_pylist():
            row_id = fts_result["_rowid"]
            results[row_id]["_score"] = fts_result["_score"]
```